trunk/src/adapt/global_href.F90

```diff
@@ -18,14 +18,21 @@ subroutine global_href
 !..collect elements
    nr_elem = NRELES
 !
+! TODO: make this routine safe to use
+!       calling break directly can result in 2-irregular mesh temporarily
+!       calling refine fixes the issue but is much more expensive
+!       if breaking elements in the right order, it should be safe to do
+!
 !..break the elements
    do i=1,nr_elem
       mdle = ELEM_ORDER(i)
       if (is_leaf(mdle)) then
          call get_isoref(mdle, kref)
          call break(mdle,kref)
+!         call refine(mdle,kref)
       endif
    enddo
+!   call close_mesh
```

Comment:
I see this is commented out but we shouldn't have to call close mesh; if we start with a 1-irregular mesh before the global h-refinement we should finish with a global h-refinement.

Reply:
You're right, `close_mesh` should not be needed here. I had tested first with putting the removal of active constrained nodes outside the subdomain into the `close_mesh` function which would be potentially cheaper than having it in `refresh` since `close_mesh` may call `refresh` multiple times. However, having it in `refresh` is the safer option if we want to enable refinements that don't call `close_mesh` at all such as the `global_href`.